build.sbt

```diff
@@ -86,21 +65,10 @@ lazy val core = crossProject(JVMPlatform, JSPlatform)
   .settings(
     name := "ip4s-core",
     libraryDependencies ++= {
-      if (isDotty.value) Nil else List("org.scala-lang" % "scala-reflect" % scalaVersion.value % "provided")
+      if (scalaVersion.value.startsWith("3")) Nil
+      else List("org.scala-lang" % "scala-reflect" % scalaVersion.value % "provided")
     },
-    scalacOptions := scalacOptions.value.filterNot(_ == "-source:3.0-migration"),
-    Compile / scalafmt / unmanagedSources := (Compile / scalafmt / unmanagedSources).value.filterNot(
-      _.toString.endsWith("Literals.scala")
-    ),
-    Test / scalafmt / unmanagedSources := (Test / scalafmt / unmanagedSources).value.filterNot(
-      _.toString.endsWith("Literals.scala")
-    ),
-    Compile / unmanagedSourceDirectories ++= {
-      val major = if (isDotty.value) "-3" else "-2"
-      List(CrossType.Pure, CrossType.Full).flatMap(
-        _.sharedSrcDir(baseDirectory.value, "main").toList.map(f => file(f.getPath + major))
-      )
-    }
+    scalacOptions := scalacOptions.value.filterNot(_ == "-source:3.0-migration")
```

Comment:
Just for my learning, why does this option not work here?

Reply:
It disables certain Scala 3 only features - in this case, the macro support.